api/src/main/java/net/kyori/adventure/text/JoinConfigurationImpl.java

```diff
@@ -144,6 +153,8 @@ public String toString() {
     final Component suffix = config.suffix();
     final Function<ComponentLike, Component> convertor = config.convertor();
     final Predicate<ComponentLike> predicate = config.predicate();
+    final Style rootStyle = config.rootStyle();
+    final boolean hasRootStyle = rootStyle != Style.empty();
 
     if (!it.hasNext()) {
       return singleElementJoin(config, null);
```

Comment:
Another reason for handling the style in the `singleElementJoin` method is so that this case can have the root style applied, which it isn't at the moment.